### ISSUE TITLE
Wlan1 not working, cannot start monitor mode

### ISSUE THREAD

**Issue by eeo1122** [0E800/kernel_flo_pie_nh#1]
Hi, I'd like to start by thanking you for bringing this project back to life.  
5 years ago i was running Kali Pwn Pad on my 2013 Nexus 7 Flo / KitKat & I was able to put the  tl-wn722n v1  Wi-Fi adapter in monitor mode and was pentesting on the go.
After my battery finally crapped out, I put it in a drawer and they it sat until 6 months ago. 

So here I am now... 

 I just successfully flashed RR-P-v7.0.1-Flo-Nethunter_r19.zip and installed nethunter according to your guide: https://github.com/0E800/Nexus-7-2013-Flo-Experimental/releases/tag/RR_PIE_Flo_Kali 

I then flashed Kernel-Nethunter-Flo-Pie-20190328_r1.zip without errors.. So far, so good.
I plugged in my Wi-Fi adapter tp-link tl-wn722n v1 into an OTG.  
  Upon opening Nethunter terminal  I selected  'Kali' shell and ran   'lsusb'  as kali@root,  yet it returns nothing.
I also tried    'ifconfig wlan1 up'   and returns  'no such device'

However, when launching Nethunter terminal and selecting  'Android-Su'  shell,  then run   'lsusb'    it returns the following:

Bus 001 Device 002:  ID  0cf3:9271           <<<  tl-wn722n
Bus 001 Device 001:  ID  1d6b:0002          <<<  OTG cable

Btw, the OTG is not powered. Is this possibly the reason the adapter doesn't respond (blinking light)?

I see in the Nethunter app under  'Kali Services'  there is Y-cable Charging.
I purchased powered Y-cable, but it only charges the tablet & doesn't put power thru to the Wi-Fi adapter.

Which Y-cable is needed, if any at all?
 
One more thing when flashing  Kernel-Nethunter-Flo-Pie-20190328_r1.zip
I see... 
Added: system/etc/firmware/htc_9271.fw

Have you experienced any of these issues yourself?
If you need me to provide additional info, please let me know.
I appreciate any help.

Thanks

**Comment by 0E800** [comment]
Happy Saturday.

Thank you for taking the time to test and leaving a detailed review of your experience.

The OTG + TPlink v.1 is fickle and frustrating to get working. I will include methods of troubleshooting and the quirks I have noticed that cause it to fail.

Before I do, please know that I had grande intensions to have this build working on N7 2012/2013, N5, and Pixel devices.
Repurposed an outdated Dell server with 80GB memory and as much hard drive space as I could get into it. 
Lots of up all night frustrations with cloning repos and trying various methods of getting kernels to compile with the various kernel mods that make Nethunter unique.

Its a tough juggle to try to get a current Android OS like O, or P, (for security updates) and then try to do what worked for Android N  and expect same results. So yeah, I got burnt out trying to finesse everything I wanted only to find mixed results getting it to work.

Then my son started 1st grade and being a full time single dad, I could not dedicate the same amount of time developing and eventually just got burnt out compiling kernels.

I do plan on revisiting this proj and still have wet dreams of making Nethunter builds that run on Android O or P and work on the N7, N5, and Pixel devices. Problem is the whole momentum towards portable pentesting devices has fizzled. Its more of a novelty and I have fallen out of the loop with regards to knowing if Binkybear or OffSec are still developing for future compatibility.

You and another user have been the only feed back I have received but its enough to make me wanna fire up the poweredge and try to pick up where I left off. I just need to get that focus that comes when progress is being made.

Anyways.. on to troubleshooting the extWifi.

Physical limitations:
The N7 2013 has a really shitty microusb port. The OTG or Y-cable you will notice wiggles like a lose tooth in the device. This causes the wifi adapter to fail whenever it wiggles. I have had better luck by wedging a piece of folded paper or foil to make the connection snug. Also used velcro to keep the wifi adapter in a fixed position to reduce wiggle.

OTG and Y adapters. I have had 50/50 luck with some cables not working. Make sure you have a few and not from same manufacturer.

 I usually test with a wired mouse to verify the usbport and otg works.

Turn the device on, wait for OS to boot, plug in tplink, goto Kali Menu and try the button for enabling wlan1.
If says device not found, unplug tplink, then plug back in and type 'dmesg' in kali terminal.

Find the USB messages when you plugged in the device. Does it recognise it as an Atheros device or does it give an error about missing firmware?
Try reflashing just the kernel or in my case, one of the various other maybe kernels and see if there is ANY change.

I noticed that turning on USB debugging can cause busybox to fail. I would try testing with it off and on. See if dmesg results change.

So verify busybox is working.

The magisk modules I include in the build were modified. I pray and sprayed the nethunter magisk mods with extra firmwares.

It possible that you may have better or worse results by installing the newer nethunter magisk modules.

I am going to make myself the user and redownload as you have and compile a step by step method to help.

You will know you are getting close when you notice that removing the TPlink causes the N7 crash and reboot.

Tldr;

Stand by I will try to make this work for you.

Thank you again for taking the time to reach out.

**Comment by eeo1122** [comment]
Good day to you,

Thank you for your prompt response.
I appreciate your dedication to the art.

A couple of things I forgot to mention...

*Flashback to 2014/2015:*
When I finally had Monitor mode working on PwnPad, I had the same tp-link
adapter I'm using now. Also the same OTG cables (2 from diff
manufacturers). I didn't have a powered y-cable. I simply connected both directly to the Micro-USB port on FLO and as soon I entered  *ifconfig
wlan1 up* the light on the adapter would start blinking.
I was getting power direct from the tablets Micro-usb port.

I used it so much that the port became very loose. Eventually it stopped
powering the adapter as well as stopped receiving a charge from the port. I was only able to charge it using a QI charger.
**but check this out...
The only function that remained was when I connected my Slim Port adapter
to HDMI, but it needed to be on the QI charging pad bar the same time.

*Flash forward to 2018*:
I purchased a new battery, as well as a new board which has the micro
usb port. The seller on Amazon called it a daughterboard, as opposed to a
Motherboard.
I recall it being marked with a different rev.#(new and improved?), but
same capabilities.
I still have the original board. If you need I'll take a pic of it and
attach for you to see.
As for the new board. I don't want to open the casing again as the plastic
snaps may break. I may be able to grab a pic from Amazon in my purchase
history. Though it may not reflect the same board I actually received.

I'm certain the OTG cable works. I can move files to usb using OTG on Two
different tablets (N7 flo & Samsung Tab E)  but I haven't tried
connecting a usb mouse or keyboard.
I will test that and let you know the results.

...also my Slim port adapter works just as the well as with the original Micro
USB port. The Slim port does require 5V and has its own Micro USB input
that also pours thru some power to the tablet. Not sure how much power but
I can tell by the battery icon it's getting a charge.
If I set the tablet on the QI charger the battery and plug power into the
slim port the tablet gets a good charge, and the image displays on the
external monitor.

I'll also do as u suggested and check dmesg output as soon as I get some
spare time to fuck with it.

Thanks again,
I hope you can follow my ramblings. Just trying to put down my thoughts as
I go.

Cheers... It's Beer-O'clock.

**Comment by eeo1122** [comment]
I'll post the results of dmesg when the adapter plugged in on my next comment, but to better understand the process of building a kernel, I did some searching and found this post by CIUFFY

https://null-byte.wonderhowto.com/how-to/android-cyanogenmod-kernel-building-monitor-mode-any-android-device-with-wireless-adapter-0162943/

It is a long read but it looks to cover the process in full.

Would you please review it and let me know I'm on right track?

Thanks.

dmesg results to come next

**Comment by eeo1122** [comment]
dmesg

Please note, I tried the following with USB debugging enabled & disabled and get same results:

I opened Nethunter Terminal & selected the Kali shell 

When first running dmesg, it printed a lot of info. So, in order to see the newly attached device(s), I first cleared the kernel ring buffer to make it easier to read by running:

dmesg -C 

Then with just the otg plugged in I ran:

dmesg

[ 3205.544586] anx7808_cbl_det_isr : detect cable removal
[ 3205.544769] msm_otg_id_pin_irq_enabled : id_pin_irq_enable = 1
[ 3205.951202] id_pin_irq_work_function: gpio_get_value(APQ_OTG_ID_PIN) = 0
[ 3205.951232] id_pin_irq_work_function: APQ_OTG_ID_PIN is low : Host mode
[ 3205.951477] msm_otg msm_otg: OTG runtime resume
[ 3205.954986] reg (HPM)
[ 3205.956420] msm_hsusb_config_vddcx: min_vol:1045000 max_vol:1320000
[ 3205.956481] msm_otg msm_otg: USB exited from low power mode
[ 3205.956878] b_idle work
[ 3205.956909] !id || id_A
[ 3205.956970] a_idle work
[ 3205.956970] !a_bus_drop && (a_srp_det || a_bus_req)
[ 3205.970703] vbus_otg is NULL.
[ 3205.970886] starting a_wait_vrise timer
[ 3206.071746] expired a_wait_vrise timer
[ 3206.071929] a_wait_vrise work
[ 3206.071960] a_vbus_vld
[ 3206.071990] msm_otg msm_otg: host on
[ 3206.072021] smb345_charger: [smb345_otg_status] otg function: on
[ 3206.080200] smb345_charger: [smb345_inok_isr] VBUS_DET = L
[ 3206.081024] msm_otg msm_otg: phy_reset: success
[ 3206.190307] Host mode: Set DC level as 0x68 for flo.
[ 3206.190826] msm_hsusb_host msm_hsusb_host: Qualcomm On-Chip EHCI Host Controller
[ 3206.192108] SELinux: initialized (dev usbfs, type usbfs), uses genfs_contexts
[ 3206.192352] msm_hsusb_host msm_hsusb_host: new USB bus registered, assigned bus number 1
[ 3206.229736] msm_hsusb_host msm_hsusb_host: irq 132, io mem 0x12500000
[ 3206.249908] msm_hsusb_host msm_hsusb_host: USB 2.0 started, EHCI 1.00
[ 3206.250396] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002
[ 3206.250793] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[ 3206.251068] usb usb1: Product: Qualcomm On-Chip EHCI Host Controller
[ 3206.251403] usb usb1: Manufacturer: Linux 3.4.113-Kali-0E800.nethunter-flo-1.0-g5452e51bb642 ehci_hcd

**The otg is seen:
[3206.250396] usb usb1: New USB device found, 
 idVendor=1d6b, idProduct=0002

-------
Here are results after connecting adapter via otg and running dmesg a second time:

+0.000244] type=1400 audit(1570458775.827:331): avc: denied { getattr } for pid=447 comm="storage[Oct 7 14:33] OTG IRQ: 132 in LPM
[  +0.000244] msm_otg msm_otg: OTG runtime resume
[  +0.000488] msm_otg msm_otg: USB exited from low power mode
[  +0.000275] portsc = 80501803
[  +0.201049] portsc = 88001205
[  +0.065033] usb 1-1: new high-speed USB device number 2 using msm_hsusb_host
[  +0.066468] portsc = 88001205
[  +0.104797] usb 1-1: New USB device found, idVendor=0cf3, idProduct=9271
[  +0.000336] usb 1-1: New USB device strings: Mfr=16, Product=32, SerialNumber=48
[  +0.000213] usb 1-1: Product: USB2.0 WLAN
[  +0.000336] usb 1-1: Manufacturer: ATHEROS
[  +0.000183] usb 1-1: SerialNumber: 12345
[  +0.014832] B_CONN set
[  +0.000030] deleting a_wait_vrise timer. remaining -56682 msec
[  +0.910004] setting pm_qos in msm_pcm_playback_copy with 1000
[  +0.000152] done setting pm_qos in msm_pcm_playback_copy

***The adapter is definitely seen upon connection.

----
Just to verify if a change is detected upon removal I unplugged the adapter and ran dmesg a third time:

[ 4823.829193] usb 1-1: USB disconnect, device number 2
[ 4823.842498] B_CONN clear
[ 4823.842559] a_host work
[ 4823.842590] !b_conn
[ 4823.842620] deleting a_wait_vrise timer. remaining -145813 msec
[ 4824.941406] setting pm_qos in msm_pcm_playback_copy with 1000
[ 4824.941711] done setting pm_qos in msm_pcm_playback_copy

***All looks good...

Please let me know if you need more info.

Cheers

**Comment by eeo1122** [comment]
I noticed from the dmesg log usb is in low power mode. I googled and found this: 

https://github.com/offensive-security/kali-nethunter/issues/1113#issuecomment-392230656

alkalinelito said they used a powered usb hub to overcome this issue.

Have you tried?

**Comment by 0E800** [comment]
Yep. Please believe me when I say I would  not have uploaded the build if I
did not confirm first that the wlan1 worked.

Ill try to have some video content of getting it to show. Sorry my friend,
work has been a bitch and im at the bottom of the bag.

I am currently working on compiling updated kernel.

Didnt want you to think I forgot and props to you for continuing.

Btw, yes, do try a batery pack with y cable.

**Comment by eeo1122** [comment]
Please don't get the idea I'm trying to rush you. I know well that work comes first.
If I find something or have questions I'll continue posting here and patiently await your reply. 
I have no formal training when it comes to writing code and compiling kernels and such but I can imagine it takes a lot of your free time.
Thanks again.

Btw my nexus 7 flo is 32gb. Not sure if that really matters but I thought it might be good to know

**Comment by 0E800** [comment]
Its takes especially long for me as I had no formal training either and im
an infant at github else I would be probably doing this for 3 diff devices
at once.

As it is, i probably only remember 1/3 of what I did when i initially
release this.

The benfit of this is when i get this down for one Rom i should be able to
create nethunter ports of the other roms as well. So we will have options.

I am mostly interested in wlan1 injection and usb hid attack. Any other
bells and whistles will have to be tested.

Be well mang .

Im available on twitter too but its mostly metaphor junkie rhetoric.

@0E800

I appreciate the real feels and I hope this endeavear

**Comment by eeo1122** [comment]
Likewise with remembering the steps from the early days. I had kept notes on my pc but the HD crashed a few years ago. I still have it just in case I find a way to recover the data. That's been on the back burner and will most likely stay there for a long while.

**Comment by 0E800** [comment]
Should have a build for you to test soon. Hopefully before tomorrow this
time.

Decided to switch to AEX AOSP Extended as the base rom as its updated to
Sep security patches and not based on Lineage. It still has the extras
similar to RRemix.

Will update the magisks versions that are bundled

Just need to apply patches to kernel and then we should be able rock on
getting a for sure solid method of ensuring the tplink works with injection.

If we see this through, everyone benefits from you making a simple
request.
Good looking out seeker 🤜

**Comment by eeo1122** [comment]
For sure. But of course all the credit goes to you for putting everything together. So many have helped me along the way, I feel obligated to give back. I'm glad to help you out as well as the community.   
I don't wanna toot my own horn but, I may have been the first to verify monitor mode working for 32gb Nexus 7 2012 (grouper) with bcmon. I posted on blogspot.com and shared with the devs. Sadly since I never backed up my G+ account, the comments are no longer visible.

http://loaded247.blogspot.com/2013/07/bcmonapk-monitor-mode-working-for.html?m=1

That was my first tablet... That's when I fell in love with android.

Anyways, I look forward to the new build.
Thanks

**Comment by eeo1122** [comment]
Back then I was such a noob that I couldn't figure out how to comment on the devs blogspot page, so I created my own account and shared my findings. My first and only blog.

**Comment by 0E800** [comment]
Okay try this build release:

https://github.com/0E800/Nexus-7-2013-Flo-Experimental/releases/tag/AEX_FLO_PIE_NH

Mostly same setup.

Make sure on latest TWRP
Boot to Recovery
Do the WIPE DATA TYPE YES 
then go back a few times and Advanced Wipe click on the boxes and wipe.
then go back a few times and click reboot / reboot to recovery
Copy AEX_Pie-v6.7-10112019-Nethunter-Flo.zip to N7
Install AEX_Pie-v6.7-10112019-Nethunter-Flo.zip
When done, reboot
After device boots up, reboot back into recovery
Install Magisk
Reboot
Open Magisk
Click on Modules
Click on the + to add module
Select KaliNethunter-v8.800
Reboot
Make sure connected to internet
Run Nethunter app and accept all the permissions
Add the CHROOT - > on SDCARD -> Minimal.
Select Kali-Linux-Nethuner (top option already selected)
Select Kali-Linux-Wireless
Click install & update
Press Enter to accept dialog box defaults
Once done reboot to recovery
Install Kernel-Nethunter-Flo-Pie-testusb.zip
Reboot
Open Nethunter app
Plug in TPlink to otg cable
Click on the upper left menu slider
Click Custom Commands
Click Wlan1 Monitor mode
Half of my my otg adapters were faulty. Test with a usb flash drive to verify the otg cables work.
I verified that I was able to get TPlink adapter to work.

Good luck. This version seems smoother

**Comment by eeo1122** [comment]
Downloading... I'll let you know my results shortly.

Thanks

**Comment by eeo1122** [comment]
I noticed your last post says to 
"Install Kernel-Nethunter-Flo-Pie-testusb.zip", yet the your latest build release

https://github.com/0E800/Nexus-7-2013-Flo-Experimental/releases/tag/AEX_FLO_PIE_NH

shows file Kernel-Nethunter-Flo-Pie-USB.zip in the Assets dropdown list.
I assume this was typo in your instructions and will install Kernel-Nethunter-Flo-Pie-USB.zip

I'll let you know what happens.

**Comment by eeo1122** [comment]
After installing/updating metapackages then rebooting to recovery. I saw Kernel-Nethunter-Flo-Pie-testusb.zip was under  /sdcard/0 which was added to the internal storage upon installing
AEX_Pie-v6.7-10112019-Nethunter-Flo.zip
I got to thinking I was wrong assuming you made a typo in your instructions...
So I tried to flash and it returned
------
Error:  Unable to mount /system as rw!
Aborting...
Failed to patch boot image!
Updater prices ever with ERROR:  1
Error installing zip file '/sdcard/0/Kernel-Nethunter-Flo-Pie-testusb.zip'
Updating partition details...
...done
------
after that error I flashed
Kernel-Nethunter-Flo-Pie-USB.zip from Assets dropdown list.
It flashed without error.
I rebooted and will continue per your instructions.

**Comment by eeo1122** [comment]
I opened Nethunter app, plugged in Tp link/otg, clicked Custom Commands, clicked the Run button for Wlan1 Monitor Mode.
Nethunter Terminal opened...

The first 2 lines show

bash:  sudo:  command not found
root@kali:/#

Looks like sudo is not installed.

I saw some text that ran quickly just before that I couldn't read. So I took a screenshot and posted on imgur.com... See link below:

<blockquote class="imgur-embed-pub" lang="en" data-id="a/YdJxDLR"><a href="//imgur.com/a/YdJxDLR">NetHunter: Wlan1 Monitor Mode error</a></blockquote>

It looks like there is error mounting kali chroot.

Another problem i found:

I ran ifconfig and it returned 
dummy0, lo & wlan0
... Still not showing wlan1

I ran iwconfig and it returned
-bash:  iwconfig:  command not found

In NetHunter, Custom Commands I long pressed Wlan1 Monitor Mode to see what's going on. 
Command:
echo ne "-\033]0;Wlan1 Monitor UP\007" && clear;sudo ifconfig wlan1 down && sudo iwconfig wlan1 mode monitor && sudo ifconfig wlan1 up && echo "wlan1 Monitor mode enabled" && sleep 3 && exit

So as I understand, the command brings down wlan1 with ifconfig in order to put in monitor mode with iwconfig and then brings wlan1 up with ifconfig...

So since chroot fails to mount and both sudo and iwconfig aren't installed the command fails.

I'm not sure what to do about chroot but I'll see what I find by installing sudo and iwconfig.

**Comment by 0E800** [comment]
I am downloading to verify.
Not sure why you got the errors.

**Comment by eeo1122** [comment]
First I ran:
apt-get update

Then to install sudo I ran:
apt-get install sudo

sudo is now installed...

For iwconfig, I reinstalled wireless-tools by running:
sudo apt-get install --reinstall wireless-tools 

iwconfig is now installed

I'll try the custom command again.

This time the terminal opens and shows
sudo:  ifconfig:  command not found
root@kali:/#

**Comment by 0E800** [comment]
You need to start over.
Somewhere root broke and messed up chroot.

Are you getting errors when you format data and wipe all the partitions in
the begining?

Lemme make sure no corrupt data with my upload and then we will figure out
what went wrong

**Comment by eeo1122** [comment]
If I try to run
ifconfig it returns as normal.. 
But if I run:  sudo ifconfig
I get this..
sudo: ifconfig: command not found

**Comment by eeo1122** [comment]
No error when formatting and wiping. I'll try after you check your upload...

Standing by.

**Comment by 0E800** [comment]
I think when you did the initial chroot install it , it didn't complete.

So far I have received no errors but I just fin installing magisk. Ill
complete the process to verify.

You shouldnt have to be trying to install packages like that. Someone is
def wrong with your install.

**Comment by 0E800** [comment]
I think what happened is, if you dont turn the screen off time to 30
minutes from the default 30 seconds the screen turns off and you lose the
view of chroot install.

Or at the step where chroot is installed and it gives you the options to
select the nethunter modules, once you click install and update it takes
you to kali terminal.
You need to do type apt-update
Then once its done close the terminal window, got back into nethunter and
redo the selecting of the modules.

You could probably fix your current build by opening nethunter and add
modules.
Just the top one and the wifi.

See the video i am adding.

**Comment by 0E800** [comment]
Please download attachment and rename it to .mp4 to view.
[ScreenRecord-2019-10-14-18-55-02.mp4.zip](https://github.com/0E800/kernel_flo_pie_nh/files/3726002/ScreenRecord-2019-10-14-18-55-02.mp4.zip)


Your build failed because you need to do this part twice in the setup

**Comment by eeo1122** [comment]
I'll try adding the modules again..

**Comment by eeo1122** [comment]
I set the display timeout to 30min and during the whole process I touched the screen to be certain the screen stayed on.

After starting the installing/updating the metapackages as you did in the video I was prompted me with some options on a bluescreen for kismet and a couple other tools.
**I didn't see that the first time**
The update continued showing a green progress bar.

At 71%  there was another prompt. 
------
Configuration file 'etc/dnsmasq.conf'
==> File on system created by you or by a script.
==> File also in package pricked by package maintainer.
    What would you like to do about it ?.....
------
I chose the default option by typing N and hitting return key which keeps the currently installed version.

Finally....that took a while.
------
Upgrade completed
Enjoy. (You can close the terminal now)
------
Closed terminal.
I wanted to be sure they were both installed so I opened NetHunter  and repeated the process. 
This time took only about 10 min. 

I closed terminal plugged in tp-link/OTG

*! The green light on adapter turns on !*😁

I opened a new terminal and ran iwconfig... wlan1 was showing.

A few sec later the tablet rebooted on its own. 😩
I let it startup with adapter still plugged in but can only get as far as swiping to unlock the screen then another reboot.😭

I'll try my other OTG cable. Might be a loose connection causing reboot.

**Comment by eeo1122** [comment]
Great work.... Btw.  Thank you. Thank you.. Thank you.

**Comment by 0E800** [comment]
Its a common issue. As soon as you start wlan1 if the cord wiggles or
becomes detached the device will crash and reboot.

Do not power on the device with the tplink attached or the tplink will
become wlan0.

There is a command you can run to safely detach the tplink but ill have to
locate it.

**Comment by 0E800** [comment]
Fyi, you will probably break your build if you try to install the Magisk Nethunter mod update via Magisk.

The included one is heavily modded by me as the new versions only include nethunter.apk nh_term.apk and VNC.apk.
I added all the other packages .

**Comment by eeo1122** [comment]
My other otg cable is connected and so far no reboot.
I tried getting monitor mode the old fashioned way by bringing down wlan1 with
------
ifconfig wlan1 down
airmon-ng start wlan1
------
*** iwconfig now shows it as wlan1mon
 
Now to bring it back up with
------
ifconfig wlan1mon up
------

ifconfig now shows
wlan1mon  MODE: Monitor

I wanted to unplug so i brought wlan1mon down and out of monitor mode then brought wlan1 back up:
ifconfig wlan1mon down
airmon-ng stop wlan1mon
ifconfig wlan1 up

I unplugged and shortly after it rebooted, but you have mentioned a reboot happening upon disconnect. 
So that's no surprise.

**Comment by 0E800** [comment]
Before unplugging tplink, try bringing wlan1 down using below command.

ip link set dev wlan1 down

Then you should be able to unplug and it wont reboot.

Usage:

# ip link set dev <interface> up
# ip link set dev <interface> down

Example:

# ip link set dev wlan1 up
# ip link set dev wlan1 down

**Comment by eeo1122** [comment]
I started monitor mode by using the custom command in Nethunter. Then tried the commands you suggested to bring it down. Still rebooted upon disconnect.

**Comment by 0E800** [comment]
I had a method, ill have to refigure it out.

**Comment by eeo1122** [comment]
Ok. I'll do some research as well. I've gotta get some stuff done. If I find anything I'll let you know. 
Till next time. 
Cheers

**Comment by lina365** [comment]
Hello eeo1122 and 0E800,

since I am a beginner regarding Nexus7 and Nethunter I need help with the installation of:
https://github.com/0E800/Nexus-7-2013-Flo-Experimental/releases/tag/AEX_FLO_PIE_NH
until now I had the Old release from 0E800 on my Nexus 7 2013:

that had an error with NMAP, etc.

After several searches and questions, 0E800 sent me the new link:


 and I read the posts from you and 0E800 here .


I also tried the new package to install but after installing Magisk and reboot only comes the Goggle "G" with loading bar and even after hours, nothing happens ...... I use the:
Instruction:
# Mostly same setup.
#Make sure on latest TWRP
#Boot to Recovery
#Do the WIPE DATA TYPE YES
#then go back a few times and Advanced Wipe click on the boxes and wipe.
#then go back a few times and click reboot / reboot to recovery
#Copy AEX_Pie-v6.7-10112019-Nethunter-Flo.zip to N7
#Install AEX_Pie-v6.7-10112019-Nethunter-Flo.zip
#When done, reboot
#After device boots up, reboot back into recovery
#Install Magisk
#Reboot
........
Then I also tried the kernel update to install and reboot but again no reaction.

In which order everything has to be installed and where will there be more stumbling blocks?

Thanks for an answer

**Comment by 0E800** [comment]
Are you certain you have Nexus 7 2013 Flo (Wifi only)?

The boot loop in which the device stays on bootanimation is usually a
kernel/ramdisk issue. Can also happen when the device isnt fully wiped and
comtains locked files or folders.

Please try downloading and flashing stock firmware from
https://developers.google.com/android/images#razor

Possibly the Android 5.1.1 will be better than the 6.0.1.  Try
https://dl.google.com/dl/android/aosp/razor-lmy48t-factory-0db13100.zip

You should also make sure to have current platform tools,
https://developer.android.com/studio/releases/platform-tools

I usually make a folder (  c:\fastboot ) and extract the files from the
platform tools folder into c:\fastboot.

Then extract the files from the Nexus 7 Firmware inside c:\fastboot  (see
image)

![Imgur](https://i.imgur.com/O9KEbsL.png)

Then plug in your phone to computer and press the power and volume down
keys until you get to the Bootloader screen.

Open command prompt: `cmd.exe`

Change to fastboot folder:   `cd \ fastboot`

Run `flash-all.bat`

This will completely wipe your device to factory defaults.

You will need to re-run the partition modify zip.

**Comment by 0E800** [comment]
Are you certain you have Nexus 7 2013 Flo (Wifi only)?

The boot loop in which the device stays on bootanimation is usually a
kernel/ramdisk issue. Can also happen when the device isnt fully wiped and
comtains locked files or folders.

Please try downloading and flashing stock firmware from
https://developers.google.com/android/images#razor

Possibly the Android 5.1.1 will be better than the 6.0.1.  Try
https://dl.google.com/dl/android/aosp/razor-lmy48t-factory-0db13100.zip

You should also make sure to have current platform tools,
https://developer.android.com/studio/releases/platform-tools

I usually make a folder (  c:\fastboot ) and extract the files from the
platform tools folder into c:\fastboot.

Then extract the files from the Nexus 7 Firmware inside c:\fastboot  (see
image)

![img](https://i.imgur.com/O9KEbsL.png)

ALT PIC: (same as above, second source)
![image](https://user-images.githubusercontent.com/6632069/68888440-ce751d00-06cf-11ea-914c-14c7eda6ac31.png)

Then plug in your phone to computer and press the power and volume down
keys until you get to the Bootloader screen.

Open command prompt: `cmd.exe`

Change to fastboot folder:   `cd \ fastboot`

Run `flash-all.bat`

This will completely wipe your device to factory defaults.

You will need to re-run the partition modify zip.

**Comment by 0E800** [comment]
Apologies, I am not very familiar with git. I will press wrong buttons sometimes.

**Comment by lina365** [comment]
Hi, thanks for the answer.
yes it is a "flo"
today i test your instructions from the last post.

After flash to: 5.1.1 the tab reboots every time. 
flashing to 6.1 not works....
this is the terminal issue:
-  Superblock backups stored on blocks:

        32768, 98304, 163840, 229376, 294912, 819200, 884736, 1605632, 2654208,
        4096000

        Allocating group tables: done
           Writing inode tables: done
       Creating journal (32768 blocks): done
       Writing superblocks and filesystem accounting information: done

       Sending 'userdata' (4272 KB)                       OKAY [  0.154s]
       Writing 'userdata'                                 FAILED (remote: 'Bogus size sparse and chunk header')
       fastboot: error: Command failed
       Press any key to exit...


which
 "You will need to re-run the partition modify zip." 
did you mean?

**Comment by lina365** [comment]
Hi,

today i test one more the installation.

1. 	unlook bootloader (flo 32G HW-Version rev_e......)
2. 	fully wiped it with stock firmware 5.1.1 (6.0.1 get some issues on the cmd Line) with the the platform tools. 
	then i can see the Android Robot with new SW inside and the Phone boots up.

What did you mean, in your las Post, with:
"You will need to re-run the partition modify zip."
after flashing in stock, after my point 2.

3. 	so i try wlan settings  and usb debugging
4.	on cmd line: adb reboot bootloader
5.	on cmd line: flash recovery twrp-3.3.1.0-flo.img
6.	on cmd line: fastboot boot twrp-3.3.1.0-flo.img
7. 	when the tab is in recovery mode
	wipe,advanced wip, reboot recovery
8.	upload AEX_Pie-v6.7 and Kernerl-Nethunter-Flo-Pie-USB.zip
9.	Install AEX_Pie-v6.7 and Kernerl-Nethunter-Flo-Pie-USB.zip
10.	reboot, can see "google" and the unlock sym then the "G" and loading bar, for a long time.......
11.	poweroff then volume key and power on
12.  	recovery mode
13	install magisk
14. 	reboot 
	can see "google" and the unlock sym then the "G" and loading bar, for a long time.......

but the tap comes not in Service.


Any Idea?

**Comment by 0E800** [comment]
After you install AEX.Nethunter.zip

There will be a file called sysrepart.zip

That will repartition your device to have larger system partition. You need
to run it every time you flash 5.1.1 stock.

So i would boot into twrp, flash sysrepart.zip

Then factory reset, FULL WIPE DATA "yes" then advanced wipe , select all
drives and wipe. Then install AeX..zip

Reboot to system, then reboot to TWRP and then install Magisk..

I will perform the same factory reset on mine so I can refresh my memory.

Think after you flash gapps and magisk then you install kalinethunter-v8.8
using magisk app.

Reboot and start Nethunter app.

Nethunter sends you to nh term to install chroot. You need to type ' apt
update ' and when its done, close the nh term app and go back to nethunter
app to start the chroot kali meta package installer again
It will take about 15 min to install and you will have to press ENTER a few
times to accept a few questions.

**Comment by lina365** [comment]
Ok, many installations, no better reaction of the system..........

the tool inside the package:
sysrepart dos not work on my pad........
lot of errors and so....

i find another tool called:
flo-deb_clamor_repartition.zip
this works.....

I can start the tab und tomorrow i will test again.

**Comment by 0E800** [comment]
My apologies, possibly the one I provided is for the 16GB version. I just
bundled together what worked for my N7tab not thinking of differences of
models.

I will be sure to include both versions in future releases and label the
files with more detail.

I re installed last night.

Wiped Full Format Yes then booted into Twrp, then installed AEXNH...zip
then immediately after before rebooting I installed the
kernel-nethunter-flo-pie-testusb.zip and then rebooted

After booting into android I rebooted back to twrp and flashed Magisk.

If you want gapps, you can try installing picogapps after aex before you
flash kernel-nethunter-testusb

**Comment by lina365** [comment]
Hi,
I did a lot of testing.
My tab has a memory problem somewhere.
I can partly AEXNH ... install and install the Kerner ... testusb comes from time to time an error that the memory is not enough.
I can still start the tab with AEX.
The intallation of the picogapps does not work.
Several attempts with repartition and other tools bring nothing.

As it looks like I will use the old version without Nmap.

  Thanks for the help, have in any case learned a lot again .......

PS: should there be another newer version I'll try it again ...... :-)

**Comment by 0E800** [comment]
I will try to play around with installing GAPPS.

Is gapps so important? I think you can install the Google Play directly to get all the functionality.

I think I recall having to customize the pico-gapps to not install a few packages to make it all fit.

I will see if I can create a more detailed (video) of the process to make it worth your while.

Thank you for assisting. I am sure others will be able to learn a thing or two without having to create an account to ask.

Be well.

**Comment by lina365** [comment]
Hi,

Since I'm still at the beginning with "Android" and "Linux Nethunter" I can only say that this is the best way to learn how to handle this thing (tablet) properly.
unlock unroot repartitionieren flashing installing and hope that it goes ........

The understanding with the processes and how it is possible to integrate the software on any system makes something addictive in any case.

Regarding the gapps, I just wanted to have a familiar handling of the taplet. Up to date I used it mostly as an e-book reader and not as the big hacker ;-) today I can already hack my own WLAN but only because I have copied the 20-digit key into the text file 🥇 
I did not want to learn or understand more ;-) ..........

I would be glad in any case if I get back a new input regarding the Nexus7 ......

Should I have more time the next days again for testing, I write again my next discoveries .....
